src/openrct2/ride/Vehicle.cpp

```diff
@@ -584,6 +584,30 @@ void Vehicle::MoveRelativeDistance(int32_t distance)
     ClearFlag(VehicleFlags::MoveSingleCar | VehicleFlags::CollisionDisabled);
 }
 
+void Vehicle::MoveToTrack(CoordsXYZ xyz, uint8_t direction, OpenRCT2::TrackElemType trackType)
+{
+    TrackLocation = xyz;
+    SetTrackDirection(direction);
+    SetTrackType(trackType);
+
+    auto curRide = GetRide();
+    if (curRide == nullptr)
+        return;
+
+    // Clip track progress to avoid being out of bounds of current piece
+    uint16_t trackTotalProgress = GetTrackProgress();
+    if (track_progress >= trackTotalProgress)
+        track_progress = trackTotalProgress - 1;
```

Comment:
The track progress is capped here but the track progress of the vehicle is not updated accordingly, so the `trackProgress` getter now returns an out of bounds value if read after updating the `trackLocation`.

Reply:
hmm looks like I didn't even validate for invalid value from `GetTrackProgress()`. Nice catch!